syntax = "proto3";

package user;

message RegisterRequest {
  string name = 1;
  string phone = 2;
  string role = 3; // rider or driver
  string password = 4;
}

message RegisterResponse {
  bool success = 1;
}

message LoginRequest {
  string phone = 1;
  string password = 2;
}

message LoginResponse {
  string token = 1;
  string role = 2;
}
message GetUserByIdRequest {
  string user_id = 1;
}
 
message GetUserByIdResponse {
  string user_id = 1;
  string name    = 2;
  string phone   = 3;
  string role    = 4;
}
service UserService {
  rpc Register(RegisterRequest) returns (RegisterResponse);
  rpc Login(LoginRequest) returns (LoginResponse);
  rpc GetUserById(GetUserByIdRequest) returns (GetUserByIdResponse);
}
